sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAPITests.groovy

```diff
@@ -424,6 +425,59 @@ class FileAPITests extends APISpec {
         assert pollResponse.getValue().getCopyId() != null
     }
 
+    def "Start copy with args fpk"() {
+        given:
+        primaryFileClient.create(1024)
+        def sourceURL = primaryFileClient.getFileUrl()
+        def filePermissionKey = shareClient.createPermission(filePermission)
+        // We recreate file properties for each test since we need to store the times for the test with getUTCNow()
+        smbProperties.setFileCreationTime(getUTCNow())
+            .setFileLastWriteTime(getUTCNow())
+            .setFilePermissionKey(filePermissionKey)
+
+        when:
+        SyncPoller<ShareFileCopyInfo, Void> poller = primaryFileClient.beginCopy(sourceURL, smbProperties, null,
+            PermissionCopyModeType.OVERRIDE, null, null, null, null)
+
+        def pollResponse = poller.poll()
+
+        then:
+        assert pollResponse.getValue().getCopyId() != null
+    }
+
+    def "Start copy with args fp"() {
+        given:
+        primaryFileClient.create(1024)
+        def sourceURL = primaryFileClient.getFileUrl()
+
+        smbProperties.setFileCreationTime(getUTCNow())
+            .setFileLastWriteTime(getUTCNow())
+
+        when:
+        SyncPoller<ShareFileCopyInfo, Void> poller = primaryFileClient.beginCopy(sourceURL, smbProperties, filePermission,
+            PermissionCopyModeType.OVERRIDE, null, null, null, null)
+
+        def pollResponse = poller.poll()
+
+        then:
+        assert pollResponse.getValue().getCopyId() != null
+    }
+
+    def "Start copy with args source and booleans"() {
+        given:
+        primaryFileClient.create(1024)
+        def sourceURL = primaryFileClient.getFileUrl()
+
+        when:
+        SyncPoller<ShareFileCopyInfo, Void> poller = primaryFileClient.beginCopy(sourceURL, null, null,
+            PermissionCopyModeType.SOURCE, true, false, null, null)
+
+        def pollResponse = poller.poll()
+
+        then:
+        assert pollResponse.getValue().getCopyId() != null
+    }
+
```

Comment:
I think these tests can be parameterized and collapsed into one test. You also shouldn't need an assert statement in a then block

Reply:
The async tests can also be collapsed

Reply:
Collapsed